Mathlib/LinearAlgebra/TensorProduct/Vanishing.lean

```diff
@@ -77,9 +77,38 @@ $$\sum_{i} a_{ij} m_i = 0.$$
 Note that this condition is not symmetric in $M$ and $N$.
 (The terminology "trivial" comes from [Stacks 00HK](https://stacks.math.columbia.edu/tag/00HK).)-/
 abbrev VanishesTrivially : Prop :=
-  ∃ (κ : Type u) (_ : Fintype κ) (a : ι → κ → R) (y : κ → N),
+  ∃ (κ : Type uκ) (_ : Fintype κ) (a : ι → κ → R) (y : κ → N),
```

Comment:
I believe that this is the problematic declaration. Lean has no way to figure out the universe `uκ` from the inputs of `VanishesTrivially`, so (as you see below) after your change it's difficult to use without you having to explicitly state which universe you're using for `uκ`. No doubt you can fix the errors by explicitly stating the universe you want to use in all the places where Lean can't figure it out (which will be everywhere it's used now). So this is probably evidence that this change is a bad idea.

Reply:
I think we shouldn't add this free universe parameter uκ to this definition; in fact I think this definition should simply use `∃ (n : Nat) (a : ι → Fin n → R) (y : Fin n → N)`, and we should add a lemma deriving the universe polymorphic Fintype version from the definition.